workflows/mhcquant.nf

```diff
@@ -100,200 +33,139 @@ include { INPUT_CHECK } from '../subworkflows/local/input_check'
-
-    // Filter by percolator q-value
-    OPENMS_IDFILTER_Q_VALUE(OPENMS_PERCOLATORADAPTER.out.idxml.flatMap { it -> [tuple(it[0], it[1], null)] })
-    ch_versions = ch_versions.mix(OPENMS_IDFILTER_Q_VALUE.out.versions.ifEmpty(null))
+    ch_versions = ch_versions.mix(OPENMS_IDMERGER.out.versions)
 
-    //
-    // SUBWORKFLOW: Refine the FDR values on the predicted subset
-    //
-    if (params.refine_fdr_on_predicted_subset && params.predict_class_1) {
-        // Run the following subworkflow
-        REFINE_FDR (
-            OPENMS_IDFILTER_Q_VALUE.out.idxml,
-            OPENMS_PSMFEATUREEXTRACTOR.out.idxml,
-            peptides_class_1_alleles
-        )
-        ch_versions = ch_versions.mix(REFINE_FDR.out.versions.ifEmpty(null))
-        // Define the outcome of the paramer to a fixed variable
-        filter_q_value = REFINE_FDR.out.filter_refined_q_value
+    // Run MS2Rescore
+    ch_clean_mzml_file
+            .map { meta, mzml -> [ groupKey([id: "${meta.sample}_${meta.condition}"], meta.group_count), mzml] }
+            .groupTuple()
+            .join(OPENMS_IDMERGER.out.idxml)
+            .map { meta, mzml, idxml -> [meta, idxml, mzml, []] }
+            .set { ch_ms2rescore_in }
+
+    MS2RESCORE(ch_ms2rescore_in)
+    ch_versions = ch_versions.mix(MS2RESCORE.out.versions)
+
+    if (params.rescoring_engine == 'percolator') {
+        // Extract PSM features for Percolator
+        OPENMS_PSMFEATUREEXTRACTOR(MS2RESCORE.out.idxml.join(MS2RESCORE.out.feature_names))
+        ch_versions = ch_versions.mix(OPENMS_PSMFEATUREEXTRACTOR.out.versions)
+
+        // Run Percolator
+        OPENMS_PERCOLATORADAPTER(OPENMS_PSMFEATUREEXTRACTOR.out.idxml)
+        ch_versions = ch_versions.mix(OPENMS_PERCOLATORADAPTER.out.versions)
+        ch_rescored_runs = OPENMS_PERCOLATORADAPTER.out.idxml
     } else {
-        // Make sure that the columns that consists of the ID's, sample names and the idXML file names are returned
-        filter_q_value = OPENMS_IDFILTER_Q_VALUE.out.idxml
+        log.warn "The rescoring engine is set to mokapot. This rescoring engine currently only supports psm-level-fdr via ms2rescore."
+        // Switch comet e-value to mokapot q-value
+        OPENMS_IDSCORESWITCHER(MS2RESCORE.out.idxml)
+        ch_versions = ch_versions.mix(OPENMS_IDSCORESWITCHER.out.versions)
+        ch_rescored_runs = OPENMS_IDSCORESWITCHER.out.idxml
     }
 
+    // Filter by percolator q-value
+    OPENMS_IDFILTER_Q_VALUE(ch_rescored_runs.map {group_meta, idxml -> [group_meta, idxml, []]})
+    ch_versions = ch_versions.mix(OPENMS_IDFILTER_Q_VALUE.out.versions)
+    ch_filter_q_value = OPENMS_IDFILTER_Q_VALUE.out.filtered
+
     //
     // SUBWORKFLOW: QUANT
     //
-    if (!params.skip_quantification) {
-        QUANT(merge_meta_map, OPENMS_PERCOLATORADAPTER.out.idxml, filter_q_value, ch_clean_mzml_file)
-        ch_versions = ch_versions.mix(QUANT.out.versions.ifEmpty(null))
+    if (params.quantify) {
```

Comment:
Same goes for all of the if-else-statements

Reply:
Same here I guess..